Networking/Networking/Remote/ProductAttributesRemote.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+/// Product Attributes: Remote Endpoints for fetching global product attributes.
+///
+public final class ProductAttributesRemote: Remote {
+
+    // MARK: - Product Attributes
+
+    /// Retrieves all of the global`ProductAttribute` available.
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll fetch remote products attributes.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    public func loadAllProductAttributes(for siteID: Int64,
+                                         completion: @escaping (Result<[ProductAttribute], Error>) -> Void) {
+
+        let path = Path.attributes
+        let request = JetpackRequest(wooApiVersion: .mark3, method: .get, siteID: siteID, path: path)
+
+        let mapper = ProductAttributeListMapper(siteID: siteID)
+
+        enqueue(request, mapper: mapper, completion: completion)
+    }
+
+    /// Create a new `ProductAttribute`.
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll add a new product attribute.
+    ///     - name: Attribute name.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    public func createProductAttribute(for siteID: Int64,
+                                       name: String,
+                                       completion: @escaping (Result<ProductAttribute, Error>) -> Void) {
+        let parameters = [
+            ParameterKey.name: name,
+            ParameterKey.slug: name,
+            ParameterKey.type: Default.type,
+            ParameterKey.orderBy: Default.orderBy,
+            ParameterKey.hasArchives: Default.hasArchives
+        ]
+
+        let path = Path.attributes
+        let request = JetpackRequest(wooApiVersion: .mark3, method: .post, siteID: siteID, path: path, parameters: parameters)
+        let mapper = ProductAttributeMapper(siteID: siteID)
+
+        enqueue(request, mapper: mapper, completion: completion)
+    }
+
+    /// Update a `ProductAttribute`.
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll update the product attribute.
+    ///     - productAttributeID: ID of the Product Attribute that will be updated.
+    ///     - name: Attribute name.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    public func updateProductAttribute(for siteID: Int64,
```

Comment:
It seems like this method updates the attribute name only? Would it make sense to make that fact explicit in the method name?

Reply:
I left it explicitly generic because in reality the user, also on wp-admin, doesn't take care of the details under the Product Attributes, since they know only the name of the Product Attributes. All the other pieces of information are internal, and not useful.